x-pack/plugins/inference/scripts/evaluation/evaluation_client.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { lastValueFrom } from 'rxjs';
+import { remove } from 'lodash';
+import type { EvaluationResult } from './types';
+import { withoutOutputUpdateEvents } from '../../common/output/without_output_update_events';
+import type { OutputAPI } from '../../common/output';
+
+export interface InferenceEvaluationClient {
+  evaluate: (input: string, criteria: string[]) => Promise<EvaluationResult>;
+  getResults: () => EvaluationResult[];
+  onResult: (cb: (result: EvaluationResult) => void) => () => void;
+}
+
+export function createInferenceEvaluationClient({
+  connectorId,
+  suite,
+  outputApi,
+}: {
+  connectorId: string;
+  suite?: Mocha.Suite;
+  outputApi: OutputAPI;
+}): InferenceEvaluationClient {
+  let currentTitle: string = '';
+  let firstSuiteName: string = '';
+
+  // get the suite name
+  if (suite) {
+    suite.beforeEach(function () {
+      const currentTest: Mocha.Test = this.currentTest!;
+      const titles: string[] = [];
+      titles.push(this.currentTest!.title);
+      let parent = currentTest.parent;
+      while (parent) {
+        titles.push(parent.title);
+        parent = parent.parent;
+      }
+      currentTitle = titles.reverse().join(' ');
+      firstSuiteName = titles.filter((item) => item !== '')[0];
+    });
+
+    suite.afterEach(function () {
+      currentTitle = '';
+    });
+  }
+
+  const onResultCallbacks: Array<{
+    callback: (result: EvaluationResult) => void;
+    unregister: () => void;
+  }> = [];
+
+  const results: EvaluationResult[] = [];
+
+  return {
+    evaluate: async (input, criteria) => {
+      const evaluation = await lastValueFrom(
+        outputApi('evaluate', {
+          connectorId,
+          system: `You are a helpful, respected assistant for evaluating task
+            inputs and outputs in the Elastic Platform.
+
+            Your goal is to verify whether the output of a task
```

Comment:
As discussed on slack, I'm afraid having a generic evaluation function for all task will be too naive. We already see with the NL-to-ESQL test suite that the evaluator gets bamboozled by the observed LLM because the evaluator has no knowledge of ESQL and no way to retrieve the documentation, so it don't spot invalid syntaxes or wrong parameters usages.

I'll be taking that one as a follow-up

Reply:
agreed, def room for improvement here